eb_hooks.py

```diff
@@ -289,6 +290,26 @@ def parse_hook_lammps_remove_deps_for_CI_aarch64(ec, *args, **kwargs):
         raise EasyBuildError("LAMMPS-specific hook triggered for non-LAMMPS easyconfig?!")
 
 
+def parse_hook_highway_handle_test_compilation_issues(ec, eprefix):
+    """
+    Solve issues with compiling or running the tests on both
+    neoverse_n1 and neoverse_v1 with Highway 1.0.4 and GCC 12.3.0:
+      - for neoverse_n1 we set optarch to GENERIC
+      - for neoverse_v1 we completely disable the tests
+    cfr. https://github.com/EESSI/software-layer/issues/469
+    """
+    if ec.name == 'Highway':
+        tcname, tcversion = ec['toolchain']['name'], ec['toolchain']['version']
+        cpu_target = get_eessi_envvar('EESSI_SOFTWARE_SUBDIR')
+        if ec.version in ['1.0.4'] and tcname == 'GCCcore' and tcversion == '12.3.0':
+            if cpu_target == CPU_TARGET_NEOVERSE_V1:
+                ec.update('configopts', '-DHWY_ENABLE_TESTS=OFF')
+            if cpu_target == CPU_TARGET_NEOVERSE_N1:
+                update_build_option('optarch', OPTARCH_GENERIC)
```

Comment:
Yikes, we have to be more careful here...

`optarch` is a general build option here, which is set for the entire EasyBuild session in which `Highway` and all missing dependencies are installed, so it's not specific to `Highway` at all. 🚨 

For this particular build, it was fine because all dependencies of `Highway` were already in place.

Reply:
We use `update_build_option` in the hook for `numpy` too, but we later revert back to the original value (see `post_single_extension_numpy`).